src/org/exist/collections/MutableCollection.java

```diff
@@ -1258,6 +1258,9 @@ private void storeXMLInternal(final Txn transaction, final DBBroker broker, fina
         if(info.isCreating()) {
             info.getTriggers().afterCreateDocument(broker, transaction, document);
         } else {
+            final StreamListener listener = broker.getIndexController().getStreamListener(document, StreamListener.ReindexMode.REPLACE_DOCUMENT);
```

Comment:
You are still mistaken, running this through the debugger shows that §startReplaceDocument` and `endReplaceDocument` happen on the same listener as `REMOVE_ALL_NODES` and `STORE`.

Just because the `listener` is set to `null` in `setMode` and `setDocument`, does not mean that you will get a new `listener`, it is entirely at the discretion of the `IndexWorker`;  None of the existing implementations of `IndexWorker` create a new listener when `getListener` is called.

Please check the code thoroughly first!